tuf_conformance/repository_simulator.py

```diff
@@ -425,16 +435,24 @@ def debug_dump(self) -> None:
         dest_dir = os.path.join(self.dump_dir, f"refresh-{self.dump_version}")
         os.makedirs(dest_dir, exist_ok=True)
 
-        for ver in range(1, len(self.signed_roots) + 1):
+        for ver in range(1, len(self.signed_mds[Root.type]) + 1):
             with open(os.path.join(dest_dir, f"{ver}.root.json"), "wb") as f:
-                f.write(self.fetch_metadata(Root.type, ver))
+                try:
+                    data = self.fetch_metadata(Root.type, ver)
+                except ValueError:
+                    data = b"No metadata found"
+                f.write(data)
 
         for role in self.mds:
             if role == Root.type:
                 continue
             quoted_role = parse.quote(role, "")
             with open(os.path.join(dest_dir, f"{quoted_role}.json"), "wb") as f:
-                f.write(self.fetch_metadata(role))
+                try:
+                    data = self.fetch_metadata(role)
+                except ValueError:
+                    data = b"No metadata found"
```

Comment:
come on, there is clearly a bug somewhere if this happens. Let's not hide that.

Looping through `self.signed_mds` instead of `self.mds` should be fine

Reply:
https://github.com/theupdateframework/tuf-conformance/pull/188/commits/ebb56b2be921fa03803750c7487a3c5d5f512f31